core/Config.php

```diff
@@ -514,7 +514,7 @@ protected function getUncached($class, $name, $sourceOptions, &$result, $suppres
 		// Include extensions only if not flagged not to, and some have been set
 		if (($sourceOptions & self::EXCLUDE_EXTRA_SOURCES) != self::EXCLUDE_EXTRA_SOURCES) {
 			// If we don't have a fresh list of extra sources, get it from the class itself
-			if (!array_key_exists($class, $this->extraConfigSources)) {
+			if (!isset($class[$this->extraConfigSources]) || $class[$this->extraConfigSources] === null) {
```

Comment:
Also planning to add the reverse, a Deoptimizer if you will, as is_array()/is_string() etc are op-code based in PHP7 if what I've read holds true.